### ISSUE TITLE
Should teleport_faint automatically heal all player monsters?

### ISSUE THREAD

**Issue by ultidonki** [Tuxemon/Tuxemon#1247]
**Is your feature request related to a problem? Please describe.**
Currently teleport_faint doesn't heal your monsters.
Modders need to add an event in each tuxemon center, that says 'if player is defeated then heal all their monsters`
Kerizane asked if we should heal all monsters automatically inside of teleport_faint. 
Does anyone object to this?

My thoughts are that I'm happy either way, honestly. But to start a discussion:
For:
1. I can't think of a reason you wouldn't heal all monsters after using teleport_faint. 
2. Adding events to every health center to automatically heal might be a bit annoying. 
Against:
1. It might be surprising to modders to have that side effect in an action. 
2. <s>If we want a cutscene or action to trigger when we teleport_faint into the cathedral, the easiest way to do it so far is to check `is player_defeated`, if automatic healing was added I don't think there's any other easy way to do this right now.</s>
We can handle this by checking the variable 'battle_lost_faint' - if it's true, then we've just fainted. So we can trigger cutscenes even if automatic healing was added inside of `teleport_faint`. (For example, if we wanted the player to lose money, or have to watch an advertisement in the center.) But nobody uses this right now and in the future there'll probably be a better solution (for example a global event that you can trigger if the player is defeated or something)

**Comment by kerizane** [comment]
You make a good point with your against argument, I think it's best not to assume too much about the game design. We can make it configurable as part of https://github.com/Tuxemon/Tuxemon/issues/1084

Open to other thoughts.

**Comment by ultidonki** [comment]
> We can make it configurable as part of https://github.com/Tuxemon/Tuxemon/issues/1084

That might be a good idea, actually. 

I'm fairly sure both mods would probably want the healing to be done automatically every time, so if it was done every time in teleport_faint, I don't think any modders would complain. Then, if someone actually thought they wanted it configurable, we could make it configurable when asked about it, but default it to 'on' so all the old mods would keep working. 

I can't think of a case where you wouldn't want to heal after using teleport_faint, so making it configurable might be a case of YAGNI too. Like I said, I'm happy either way, and I think both campaigns will be too, so it's up to everyone else to decide.

**Comment by ultidonki** [comment]
> If we want a cutscene or action to trigger when we teleport_faint into the cathedral, the easiest way to do it so far is to check is player_defeated, if automatic healing was added I don't think there's any other easy way to do this right now.

I've just found out there is an easy way to do this - we can handle this by checking the variable 'battle_lost_faint' - if it's true, then we've just fainted. So we can trigger cutscenes even if automatic healing was added inside of teleport_faint. That makes me happier about doing it if everyone else agrees.

**Comment by kerizane** [comment]
Let's add healing to the teleport_faint action. It's what most people will want I think.

**Comment by ultidonki** [comment]
> Let's add healing to the teleport_faint action. It's what most people will want I think.

Sounds good to me, I don't hear any complaints, and I think it makes sense. :smiley:

**Comment by JaskRendix** [comment]
I was doing some tests and this works, but there are two problems.
```
        for monster in player.monsters:
            if player.game_variables["battle_last_result"] == "lost":
                monster.current_hp = monster.hp
                monster.status = list()
```
add the code below **self.session.client.event_engine.execute_action("teleport", teleport)**.

It works: faint -> center, but:
1. if you look at the monsters menu, the bar is red (wrong, because the hp are restored);
2. if you enter in a map with "teleport_faint" / "is player_defeated" -> you're stuck inside
3. you're stuck in the route, you fight, you lose, you get teleported in the center -> you're stuck inside the center

It looks like transition_teleport is gone

if you:
1. a. you heal again your monsters in the center -> you're not stuck

**Comment by kerizane** [comment]
Can you create a draft PR for this? It would make it easier to understand and discuss your change.

**Comment by JaskRendix** [comment]
Sure

**Comment by JaskRendix** [comment]
it has been fixed back in September #1312, eventually it can be closed.
